x-pack/plugins/security_solution/public/resolver/store/data/selectors.ts

```diff
@@ -301,3 +301,48 @@ export function databaseDocumentIDToAbort(state: DataState): string | null {
     return null;
   }
 }
+
+/**
+ * Return something? for a process.
+ * TODO, how does this work by taking entity id? what does it take when we want to use it w/ a legacy process event
+ */
+const relatedEventStatsForProcess: (
+  state: DataState
+) => (event: ResolverEvent) => ResolverNodeStats | null = createSelector(
+  relatedEventsStats,
+  (statsMap) => {
+    if (!statsMap) {
+      return () => null;
+    }
+    return (event: ResolverEvent) => {
+      const nodeStats = statsMap.get(uniquePidForProcess(event));
+      if (!nodeStats) {
+        return null;
+      }
+      return nodeStats;
+    };
+  }
+);
+
+/**
+ * Return the 'total' stats (whatever that means) for a process by entity ID
+ * TODO, how does this work for legacy events?
+ */
+export const relatedEventTotalForProcess: (
+  state: DataState
+) => (event: ResolverEvent) => number | null = createSelector(
+  relatedEventStatsForProcess,
+  (statsForProcess) => {
+    return (event: ResolverEvent) => {
+      const stats = statsForProcess(event);
+      if (!stats) {
+        return null;
+      }
+      let total = 0;
+      for (const value of Object.values(stats.events.byCategory)) {
```

Comment:
Is `events.byCategory` a structure we made up?

Reply:
`events.byCategory` is what the backend returns. It's an object of number of events that fall in a particular category for example:

```
{
  'network': 3,
  'file': 2,
}
```